google/cloud/aiplatform/base.py

```diff
@@ -0,0 +1,110 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import abc
+from typing import Optional
+
+from google.auth import credentials as auth_credentials
+from google.cloud.aiplatform import utils
+from google.cloud.aiplatform import initializer
+
+
+class AiPlatformResourceNoun(metaclass=abc.ABCMeta):
```

Comment:
Should we have unit tests for this class?

Reply:
We cannot instantiate this class so we would need to subclass it and would be duplicating implementation and testing done by the Model class.

Reply:
Got it, thanks